src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs

```diff
@@ -82,36 +81,62 @@ private void EmitGetCoreMethod()
-                                throw new {{Identifier.InvalidOperationException}}();
+                                EmitCastToIConfigurationSection();
+                                EmitBindingLogic(
+                                    stringParsableType,
+                                    Expression.sectionValue,
+                                    Expression.sectionPath,
+                                    writeOnSuccess: parsedValueExpr => _writer.WriteLine($"return {parsedValueExpr};"),
+                                    checkForNullSectionValue: stringParsableType.StringParsableTypeKind is not StringParsableTypeKind.AssignFromSectionValue,
+                                    useIncrementalStringValueIdentifier: false);
                             }
-                            """);
-
-                        EmitBindLogicFromString(
-                            stringParsableType,
-                            Expression.sectionValue,
-                            Expression.sectionPath,
-                            writeOnSuccess: parsedValueExpr => _writer.WriteLine($"return {parsedValueExpr};"),
-                            checkForNullSectionValue: stringParsableType.StringParsableTypeKind is not StringParsableTypeKind.AssignFromSectionValue,
-                            useIncrementalStringValueIdentifier: false);
-                    }
-                    else if (!EmitInitException(effectiveType))
-                    {
-                        EmitBindCoreCall(effectiveType, Identifier.instance, Identifier.configuration, InitializationKind.Declaration);
-                        _writer.WriteLine($"return {Identifier.instance};");
+                            break;
+                        case ConfigurationSectionSpec configurationSectionSpec:
+                            {
+                                EmitCastToIConfigurationSection();
+                                _writer.WriteLine($"return {Identifier.section};");
+                            }
+                            break;
+                        case ComplexTypeSpec complexType:
+                            {
+                                if (complexType.CanInstantiate)
+                                {
+                                    EmitBindingLogic(complexType, Identifier.instance, Identifier.configuration, InitializationKind.Declaration);
+                                    _writer.WriteLine($"return {Identifier.instance};");
+                                }
+                                else if (type is ObjectSpec { InitExceptionMessage: string exMsg })
+                                {
+                                    _writer.WriteLine($@"throw new {Identifier.InvalidOperationException}(""{exMsg}"");");
+                                }
+                                else
+                                {
+                                    _writer.WriteLine($"// Cannot create an instance of {type.DisplayString}.");
```

Comment:
>_writer.WriteLine($"// Cannot create an instance of {type.DisplayString}."); [](http://example.com/codeflow?start=36&length=76)

how this is useful to emit in the code?

Reply:
Added it for UX but the "not supported" state should be clear from looking at the empty code block. I'll remove it.